cpp/include/cudf/copying.hpp

```diff
@@ -44,22 +44,24 @@ gdf_column empty_like(gdf_column const& input);
  * @brief Allocates a new column of the same size and type as the input.
  *
  * @param input The input column to emulate
- * @param mask_alloc Determines under what condition to allocate the mask. Defaults to RETAIN.
+ * @param mask_alloc Policy for allocating null mask. Defaults to RETAIN.
  * @param stream Optional stream in which to perform copies
  * @return gdf_column An allocated column of same size and type of input
  */
-gdf_column allocate_like(gdf_column const& input, MaskAlloc mask_alloc = MaskAlloc::RETAIN, cudaStream_t stream = 0);
+gdf_column allocate_like(gdf_column const& input, mask_allocation_policy mask_alloc = mask_allocation_policy::RETAIN,
```

Comment:
No need to qualify every RETAIN with `mask_allocation_policy::`. it makes lines like this one too long (at least wrap the line), and it turns a lot of would-be one-liners into multi-line expressions.

Reply:
I was unaware. The compiler infers the type? Or am I missing something...

Reply:
This is an unscoped enum, which puts its values into the surrounding scope (`namespace cudf` in this case). C++11 added `enum class`, which requires the kind of scoping you are using. However Cython doesn't support it yet, so we can't use it in cuDF public APIs like this one.

Reply:
To answer your question, I don't think there's a type to infer for an unscoped enum. its values just have integral type.